src/Http/Error.php

```diff
@@ -30,17 +30,28 @@
 class Error extends HtmlResponse
 {
     /**
-     * Takes the value $val and converts it to a PSR7 StreamInterface.
+     * Takes the status code and and use the Zend\Response constructor to provide
+     * the approcriate reason phrase. It also add the appropriate body using
+     * smarty templates based on status code.
      *
-     * @param string $val The string value to convert to a stream.
+     * @param ServerRequestInterface $request this HTTP request.
+     * @param int                    $status  The HTTP status code to use.
+     * @param string                 $message A value to pass to smarty template
      */
-    public function __construct(ServerRequestInterface $request, int $status, string $message = '')
-    {
-        // TODO :: Extract baseurl from request. The '/' might break if the DirectoryRoot is different
-        $tpl_data = array(
-                     'message' => $message,
-                     'baseurl' => '/',
-                    );
+    public function __construct(
+        ServerRequestInterface $request,
+        int $status,
+        string $message = ''
+    ) {
+
+        $uri     = $request->getURI();
+        $baseurl = @strstr($uri->__toString(), $uri->getPath(), true);
```

Comment:
why is this suppressing the error message?

Reply:
I was getting `PHP Warning:  strstr(): Empty needle` when no path was provided. I could check for empty string befor the call.

Reply:
I think that's better than suppressing the error, otherwise if another error/deprecation/whatever creeps into the line later or the warning gets upgraded to a fatal error we won't notice until it's too late